src/Tasks/Microsoft.NET.Build.Tasks/EmbedAppNameInHost.cs

```diff
@@ -59,7 +60,24 @@ protected override void ExecuteCore()
             {
                 Directory.CreateDirectory(destinationDirectory);
             }
-            File.WriteAllBytes(ModifiedAppHostPath, array);
+
+            if (RuntimeInformation.IsOSPlatform(OSPlatform.Windows))
```

Comment:
OK I will remove the 'if', and log a future issue for the performance aspect so we only change the marker and not the whole file. The apphost file is ~77K so fairly small, but still we should do the right thing here.